aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SignInCustomActions.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.actions
+
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.AuthFlowType
+import aws.sdk.kotlin.services.cognitoidentityprovider.model.ChallengeNameType
+import com.amplifyframework.auth.AuthException
+import com.amplifyframework.auth.cognito.AuthConstants.KEY_SECRET_HASH
+import com.amplifyframework.auth.cognito.AuthConstants.KEY_USERNAME
+import com.amplifyframework.auth.cognito.AuthEnvironment
+import com.amplifyframework.auth.cognito.helpers.AuthHelper
+import com.amplifyframework.auth.cognito.helpers.SignInChallengeHelper
+import com.amplifyframework.statemachine.Action
+import com.amplifyframework.statemachine.codegen.actions.CustomSignInActions
+import com.amplifyframework.statemachine.codegen.events.AuthenticationEvent
+import com.amplifyframework.statemachine.codegen.events.CustomSignInEvent
+
+object SignInCustomActions : CustomSignInActions {
+
+    override fun initiateCustomSignInAuthAction(event: CustomSignInEvent.EventType.InitiateCustomSignIn): Action =
+        Action<AuthEnvironment>("InitCustomAuth") { id, dispatcher ->
+            logger?.verbose("$id Starting execution")
+            val evt = try {
+                val secretHash = try {
+                    AuthHelper.getSecretHash(
+                        event.username,
+                        configuration.userPool?.appClient,
+                        configuration.userPool?.appClientSecret
+                    )
+                } catch (e: java.lang.Exception) {
+                    null
+                }
+
+                var authParams = mapOf(KEY_USERNAME to event.username)
+                secretHash?.also { authParams = authParams.plus(KEY_SECRET_HASH to secretHash) }
```

Comment:
I know we've called it out in a few cases, but wanted to make sure none get missed. Used val mutableMap here, and also use 'it' on secretHash

Reply:
done